boards/nxp/frdm_mcxn947/board.cmake

```diff
@@ -4,10 +4,14 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-if(CONFIG_SOC_MCXN947_CPU0)
+if(CONFIG_SOC_MCXN947_CPU0 OR CONFIG_SECOND_CORE_MCUX)
   board_runner_args(jlink "--device=MCXN947_M33_0" "--reset-after-load")
   board_runner_args(linkserver  "--device=MCXN947:FRDM-MCXN947")
-  board_runner_args(linkserver  "--core=cm33_core0")
+  if (CONFIG_SECOND_CORE_MCUX)
```

Comment:
i have been told we should put no space after if and between parentheses in cmake ie `if()` not `if ()`

Reply:
@nordicjm blocked one of my PR about this today is why I mention it

Reply:
> I don't think this is a rule. There are hundreds of examples of this format in the cmake hierarchy. Although I would say that there should be consistency in the same file.

Something being added as wrong doesn't mean it can be treated as right, there should be no spaces between the opcode and bracket in cmake

Reply:
time to fix the tree (quick grep shows at least 347 occurances of `if ()`, enable cmakelint in compliance check, sit back and watch people fighting bots :)

Reply:
for reference:

https://pypi.org/project/cmakelint/


and it does catch this case:

```CMakeLists.txt:320: Extra spaces between 'if' and its () [whitespace/extra]```

Reply:
This is planned though using cmake-format as per https://github.com/zephyrproject-rtos/zephyr/pull/82839